### PR TITLE
Simplify generating the HTML for the word so far

### PR DESCRIPTION
**Pull request by urkerab**
Tagging @bumbadadabum because I don't know why the `<small>` is needed; it looks the same to me either way. Alternatively, we can remove the spaces entirely and fiddle with the CSS e.g. `letter-spacing:0.333ch`.

### CODE REVIEW

**Comment by bumbadadabum** [comment]
That was zarel, not me

File: chat-plugins/hangman.js [hunk 1]
```diff
@@ -135,19 +135,8 @@ class Hangman extends Rooms.RoomGame {
...
-					wordString += '<font color="#7af87a">' + this.word[i] + '</font>';
-				} else {
-					wordString += this.wordSoFar[i];
-				}
-			}
-		} else {
-			wordString = this.wordSoFar.join('<small>&nbsp;</small>');
-		}
+		let wordString = this.wordSoFar.join('&nbsp;');
+		if (result === 1) wordString = wordString.replace(/_/g, '<font color="#7af87a">_</font>');
```
Comment (Zarel) [inline chat-plugins/hangman.js]: This doesn't do the same thing at all; the point is that we're supposed to replace `_` with the actual letter that belongs there if the result is `1` (i.e. if the game ends, possibly in a failure)

**Comment by Zarel** [comment]
Something like `margin` or `padding` would also work in place of `letter-spacing`

File: chat-plugins/hangman.js (hunk 1 above)
Comment (urkerab) [inline chat-plugins/hangman.js]: Sorry I'm blind...

**Comment by urkerab** [comment]
> Something like margin or padding would also work in place of letter-spacing

Wouldn't that still require tags around each letter? `letter-spacing` avoids that.

**Comment by Zarel** [comment]
> Wouldn't that still require tags around each letter? `letter-spacing` avoids that.

It's IE6-compatible, though. :P

Honestly though, I'm fine with `letter-spacing`; I'm mostly sure it's compatible with anything we support.

**Comment by Zarel** [comment]
So......

**Comment by urkerab** [comment]
I finally got around to rewriting this using letter-spacing. And I now show the failed letters correctly. And with the minimum of additional HTML.

File: chat-plugins/hangman.js [hunk 2]
```diff
@@ -135,22 +135,16 @@ class Hangman extends Rooms.RoomGame {
...
-			wordString = this.wordSoFar.join('<small>&nbsp;</small>');
+			let word = this.word;
+			wordString = wordString.replace(/_+/g, function (match, offset) {
+				return '<font color="#7af87a">' + word.substr(offset, match.length) + '</font>';
+			});
 		}
 
 		if (this.hint) output += '<div>(Hint: ' + Tools.escapeHTML(this.hint) + ')</div>';
-		output += '<p style="font-weight:bold;font-size:12pt">' + wordString + '</p>';
+		output += '<p style="font-weight:bold;font-size:12pt;letter-spacing:1ex">' + wordString + '</p>';
```
Comment (Zarel) [inline chat-plugins/hangman.js]: What kind of unit is `ex`?

Comment (urkerab) [inline chat-plugins/hangman.js]: Half an `em`, I think. I'm sure it could be replaced with something like `6pt` though.

Comment (Zarel) [inline chat-plugins/hangman.js]: Spec says the x-height of the font, which is usually approximately half an `em`. (Contrast an `en`, which I think is exactly half an `em`). I'd probably prefer a more common unit, yes.

**Comment by urkerab** [comment]
Changed to 3pt as that looks (to me) closest to the previous display.

**Pull request merged**